java-checks/src/main/java/org/sonar/java/checks/UtilityClassWithPublicConstructorCheck.java

```diff
@@ -80,17 +80,22 @@ private static boolean extendsAnotherClassOrImplementsSerializable(ClassTree cla
   }
 
   private static boolean hasOnlyStaticMembers(ClassTree classTree) {
-    if (classTree.members().isEmpty()) {
+    List<Tree> members = classTree.members();
+    if (members.isEmpty() || noStaticMember(members)) {
```

Comment:
`members.isEmpty()` is redundant here, because `noStaticMember` will also be true when there are no classes.
Whole method can be perhaps replaced if logic is cleaned up. We need to test that when there is at least one member, which is not constructor nor empty statement, all of them are static.
```
private static boolean hasOnlyStaticMembers(ClassTree classTree) {
    List<Tree> members = classTree.members().stream()
      .filter(m -> !m.is(Tree.Kind.CONSTRUCTOR, Tree.Kind.EMPTY_STATEMENT))
      .collect(Collectors.toList());
    return !members.isEmpty() && members.stream().allMatch(UtilityClassWithPublicConstructorCheck::isStatic);
  }
```